apps/research-service/src/site-polygons/site-polygons.controller.ts

```diff
@@ -98,25 +107,23 @@ export class SitePolygonsController {
     if (!query.includeTestProjects && query.siteId == null && query.projectId == null) {
       await queryBuilder.excludeTestProjects();
     }
-    if (query.search) {
+    if (query.search != null) {
       await queryBuilder.addSearch(query.search);
     }
-    const document = buildJsonApi(SitePolygonDto, { pagination: isNumberPage(query.page) ? "number" : "cursor" });
+    const document = buildJsonApi(SitePolygonFullDto, { pagination: isNumberPage(query.page) ? "number" : "cursor" });
     for (const sitePolygon of await queryBuilder.execute()) {
       const indicators = await this.sitePolygonService.getIndicators(sitePolygon);
-      const establishmentTreeSpecies = await this.sitePolygonService.getEstablishmentTreeSpecies(sitePolygon);
-      const reportingPeriods = await this.sitePolygonService.getReportingPeriods(sitePolygon);
-      document.addData(
-        sitePolygon.uuid,
-        new SitePolygonDto(
-          sitePolygon,
-          sitePolygon.polygon?.polygon,
-          indicators,
-          establishmentTreeSpecies,
-          reportingPeriods,
-          sitePolygon.site?.name
-        )
-      );
+      if (lightResource) {
+        document.addData(sitePolygon.uuid, new SitePolygonLightDto(sitePolygon, indicators));
+      } else {
+        const establishmentTreeSpecies = await this.sitePolygonService.getEstablishmentTreeSpecies(sitePolygon);
+        const reportingPeriods = await this.sitePolygonService.getReportingPeriods(sitePolygon);
+
+        document.addData(
+          sitePolygon.uuid,
+          new SitePolygonFullDto(sitePolygon, indicators, establishmentTreeSpecies, reportingPeriods)
```

Comment:
It seems like this controller should be using the `buildLightDto` / `buildFullDto` methods from the service instead of creating the DTOs itself?